### PR TITLE
wamr-python: Enable debugging WASM and grant dir access

### PR DESCRIPTION
**Pull request by tonibofarull**
Enables debugging a WASM loaded and executed from Python.
Exposes API to enable access to list of host directories. Similar to `--dir` in `iwasm`.

### CODE REVIEW

File: language-bindings/python/src/wamr/wamrapi/wamr.py [hunk 1]
```diff
@@ -129,21 +179,53 @@ def _create_module_inst(self, module: Module, stack_size: int, heap_size: int) -
         return module_inst
 
 
+id_to_exec_env_mapping = {}
+
+
 class ExecEnv:
-    def __init__(self, module_inst: Instance, stack_size: int = 65536):
+    def __init__(self, module_inst: Instance | None = None, stack_size: int = 65536):
```
Comment (lum1n0us) [inline language-bindings/python/src/wamr/wamrapi/wamr.py:186]: I am not sure `module_inst` is `None`able. It should be quite critical. Please make more tests

File: language-bindings/python/src/wamr/wamrapi/wamr.py [hunk 2]
```diff
@@ -129,21 +179,53 @@ def _create_module_inst(self, module: Module, stack_size: int, heap_size: int) -
...
+        self.own_c = True
+
+        id_to_exec_env_mapping[str(addressof(self.exec_env.contents))] = self
 
     def __del__(self):
-        print("deleting ExecEnv")
-        wasm_runtime_destroy_exec_env(self.exec_env)
+        if self.own_c:
+            print("deleting ExecEnv")
+            wasm_runtime_destroy_exec_env(self.exec_env)
```
Comment (wenyongh) [inline language-bindings/python/src/wamr/wamrapi/wamr.py:196]: Should we also remove related element from id_to_exec_env_mapping?

Comment (tonibofarull) [reply language-bindings/python/src/wamr/wamrapi/wamr.py:201]: done here: https://github.com/bytecodealliance/wasm-micro-runtime/pull/2449/commits/e224b300a9843319abae63ea1de1c489161f74ee

**Review by wenyongh — APPROVED** [review]
LGTM

**Pull request merged**